src/fu-engine.c

```diff
@@ -5017,6 +5020,188 @@ fu_engine_get_host_machine_id (FuEngine *self)
+		return;
+	}
+	releases = fu_engine_get_releases_for_device (self, device, &error_local);
+	if (releases == NULL) {
+		g_debug ("no releases for main-system-firmware: %s",
+			 error_local->message);
+		return;
+	}
+
+	/* check the age */
+	for (guint i = 0; i < releases->len; i++) {
+		FwupdRelease *rel_tmp = g_ptr_array_index (releases, i);
+		if (rel_newest == NULL ||
+		    fwupd_release_get_created (rel_tmp) > fwupd_release_get_created (rel_newest))
+			rel_newest = rel_tmp;
+	}
+	if (now - fwupd_release_get_created (rel_newest) > 60 * 60 * 24 * 30 * 12) {
+		g_debug ("newest release is %" G_GUINT64_FORMAT " months old",
+			 (now - fwupd_release_get_created (rel_newest)) / (60 * 60 * 24 * 30));
+		return;
+	}
+
+	/* add runtime attribute */
+	attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.Updates");
+	fwupd_hsi_attr_set_name (attr, "Firmware Supported");
+	fwupd_hsi_attr_set_summary (attr, "There is firmware less than 12 months old");
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_SUCCESS);
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_UPDATES);
+	g_ptr_array_add (attrs, attr);
+
+	/* do we have attestation checksums */
+	if (fwupd_release_get_checksums(rel_newest)->len > 0) {
+		attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.Attestation");
+		fwupd_hsi_attr_set_name (attr, "Firmware Attestation");
+		fwupd_hsi_attr_set_summary (attr, "Firmware has attestation checksums");
+		fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_SUCCESS);
+		fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_ATTESTATION);
+		g_ptr_array_add (attrs, attr);
+	}
+}
+
+GPtrArray *
+fu_engine_get_host_security_attrs (FuEngine *self, GError **error)
+{
+	GPtrArray *plugins = fu_plugin_list_get_all (self->plugin_list);
+	g_autoptr(GPtrArray) hsi_attrs = NULL;
+
+	/* built in */
+	hsi_attrs = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
+	fu_engine_add_hsi_attrs_tainted (self, hsi_attrs);
+	fu_engine_add_hsi_attrs_supported (self, hsi_attrs);
+
+	/* call into plugins */
+	for (guint j = 0; j < plugins->len; j++) {
+		FuPlugin *plugin_tmp = g_ptr_array_index (plugins, j);
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_plugin_runner_add_hsi_attrs (plugin_tmp,
+						     hsi_attrs,
+						     &error_local)) {
+			g_warning ("failed to add HSI attrs for %s: %s",
+				   fu_plugin_get_name (plugin_tmp),
+				   error_local->message);
+			continue;
+		}
+	}
+
+	/* sanity check */
+	if (hsi_attrs->len == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "no host security attributes");
+		return NULL;
+	}
+	return g_steal_pointer (&hsi_attrs);
+}
+
+static gchar *
+fu_engine_hsi_attrs_to_string (GPtrArray *hsi_attrs)
+{
+	guint hsi_number = 0;
+	FwupdHsiAttrFlags flags = FWUPD_HSI_ATTR_FLAG_NONE;
+	GString *str = g_string_new ("HSI:");
+
+	/* find the highest HSI number where there are no failures and at least
+	 * one success */
+	for (guint j = 1; j < 9; j++) {
```

Comment:
Why did you offer 9 as the upper boundary?  Shouldn't this be 5?

Reply:
Maybe we want something like `FWUPD_HSI_ATTR_NUMBER_MAX` exported in fwupd-hsi-attr.h?

Reply:
This is what I did.